modules/core/shared/src/main/scala/gem/util/Format.scala

```diff
@@ -4,16 +4,28 @@
 package gem.util
 
 import cats.arrow._
+import cats.implicits._
 import monocle.{ Iso, Prism }
+import scala.reflect.runtime.universe.TypeTag
 
 /**
  * A normalizing optic, isomorphic to Prism but with different laws, specifically `getOption`
  * need not be injective; i.e., distinct inputs may have the same getOption result, which combined
  * with a subsequent `reverseGet` yield a normalized form for A. Composition with stronger optics
  * (`Prism` and `Iso`) yields another `Format`.
  */
+@SuppressWarnings(Array("org.wartremover.warts.Null"))
 final case class Format[A, B](getOption: A => Option[B], reverseGet: B => A) {
 
+  /** Like getOption, but throws IllegalArgumentException on failure. */
+  @SuppressWarnings(Array("org.wartremover.warts.DefaultArguments", "org.wartremover.warts.Throw"))
+  def unsafeGet(a: A)(implicit ev: TypeTag[B] = null): B =
+    getOption(a).getOrElse {
+      // If a TypeTag is available we can construct a more useful message.
+      val prefix = Option(ev).foldMap(tag => s"${tag.tpe}: ")
```

Comment:
nice

Reply:
`foldMap` on `Option` ends up being really useful.

Reply:
Nothing is being appended so it's just a shorter version of

```
Option(ev).map(tag => s"${tag.tpe}: ").orEmpty
```
 ?

Reply:
For `Option` yes. More generally it's `foldRight` with the monoid identity as zero.